plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -491,7 +499,7 @@ else if (partitionKeys.containsKey(column.getId())) {
                             deserializePartitionValue(trinoType, partitionKeys.get(column.getId()).orElse(null), column.getName()))));
                 }
                 else if (column.isPathColumn()) {
-                    columnAdaptations.add(ColumnAdaptation.constantColumn(nativeValueToBlock(FILE_PATH.getType(), utf8Slice(path.toString()))));
```

Comment:
Q are we changing the behavior here? i.e. before this PR, effectively `hadoopPath()` was being returned here since that was propagated from the split, but now it's the actual path without encoding.

Reply:
Yes, it's changing. Relates to https://github.com/trinodb/trino/pull/13012#discussion_r909434531